modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -439,29 +474,21 @@ GDScriptParser::DataType GDScriptAnalyzer::resolve_datatype(GDScriptParser::Type
 	StringName first = p_type->type_chain[0]->name;
 
 	if (first == SNAME("Variant")) {
-		result.kind = GDScriptParser::DataType::VARIANT;
 		if (p_type->type_chain.size() > 1) {
-			push_error(R"("Variant" type don't contain nested types.)", p_type->type_chain[1]);
-			return GDScriptParser::DataType();
+			// TODO: Variant does actually have a nested Type though.
+			push_error(R"(Variant doesn't contain nested types.)", p_type->type_chain[1]);
+			return bad_type;
 		}
-		return result;
```

Comment:
i was very annoyed that Variant and Object were the only ones to return a good result early
removing this actually fixes an annoying bug that allows the type `Variant[int]` but chops off the `[int]` part in the analyzer. same with Object

Reply:
Yeah, deleted my comment after realizing that you can probably still fill out some stuff later, so you can do all these checks and do some partial fills :)

Reply:
:)